### PR TITLE
III-3851 Refactor Natural value object

### PR DESCRIPTION
**Pull request by LucWollants**
### Added
- Added custom abstract `Natural` class
- Added custom `Delay` class
- Added custom `Age` class
### Changed
- Refactor `Price` class to use custom abstract `Natural` class
---
Ticket: https://jira.uitdatabank.be/browse/III-3851

### CODE REVIEW

File: tests/NaturalTest.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            [99]
+        );
+
+        $this->assertEquals('99', $natural->toString());
+    }
+
+    /**
+     * @test
+     */
+    public function it_return_the_value(): void
```
Comment (jdrieghe) [inline tests/NaturalTest.php:41]:
```suggestion
    public function it_can_return_the_value(): void
```

**Review by jdrieghe — APPROVED** [review]

File: src/ElasticSearch/Offer/ElasticSearchOfferQueryBuilder.php [hunk 2]
```diff
@@ -154,7 +154,7 @@ public function withDateRangeFilter(
 
     public function withLocalTimeRangeFilter(int $localTimeFrom = null, int $localTimeTo = null): self
     {
-        $this->guardNaturalIntegerRange('localTime', new Natural($localTimeFrom), new Natural($localTimeTo));
+        $this->guardNaturalIntegerRange('localTime', new Age($localTimeFrom), new Age($localTimeTo));
```
Comment (bertramakers) [inline src/ElasticSearch/Offer/ElasticSearchOfferQueryBuilder.php:157]: I don't think this is right?

File: src/ElasticSearch/Offer/ElasticSearchOfferQueryBuilder.php [hunk 3]
```diff
@@ -182,7 +182,7 @@ public function withSubEventFilter(SubEventQueryParameters $subEventQueryParamet
         $this->guardDateRange('date', $from, $to);
 
         if ($localTimeFrom && $localTimeTo) {
-            $this->guardNaturalIntegerRange('localTime', new Natural($localTimeFrom), new Natural($localTimeTo));
+            $this->guardNaturalIntegerRange('localTime', new Age($localTimeFrom), new Age($localTimeTo));
```
Comment (bertramakers) [inline src/ElasticSearch/Offer/ElasticSearchOfferQueryBuilder.php:185]: This should also not be `Age`

**Comment by LucWollants** [comment]
> @LucWollants can you create a follow-up PR with fixes for my comments above? Thanks!

Fixed inside https://github.com/cultuurnet/udb3-search-service/pull/107

**Pull request merged**